generators/core.go

```diff
@@ -10,81 +10,97 @@ import (
 	"text/template"
 )
 
-const preamble = `// NOTICE
-// This file was automatically generated by ../generators/core.go. Do not edit it!
-// You can regenerate it by running 'go generate ./core' from the directory above.
-
-package core
-
-import (
-	"github.com/crabmusket/gosunspec/smdx"
-)
-`
-
-const model = `
+const file = `
 {{define "point"}}{{.Id}} {{.Type|goType}} ` + "`" + `sunspec:"offset={{.Offset}}{{if gt .Length 0}},len={{.Length}}{{end}}{{if ne .ScaleFactor ""}},sf={{.ScaleFactor}}{{end}}{{if ne .Access ""}},access={{.Access}}{{end}}"` + "`" + `{{end}}
-{{$model := .Model }}
+{{define "model"}}
+{{$model := index .Models 0 }}
+{{$strings := index .Strings 0 }}
 
-// Block{{.Model.Id}} - {{.Strings.ModelStrings.Label}} - {{.Strings.ModelStrings.Description}}
-{{range (repeatingBlocks .Model)}}
+// Block{{$model.Id}} - {{$strings.ModelStrings.Label}} - {{$strings.ModelStrings.Description}}
+{{range (repeatingBlocks $model)}}
 type Block{{$model.Id}}Repeat struct {
 	{{range .Points}}{{template "point" .}}
 {{end}}
 }
 {{end}}
 
-type Block{{.Model.Id}} struct {
-{{range (fixedBlocks .Model)}}
+type Block{{$model.Id}} struct {
+{{range (fixedBlocks $model)}}
 {{range .Points}}    {{template "point" .}}
 {{end}}
 	{{end}}
-{{range (repeatingBlocks .Model)}}
+{{range (repeatingBlocks $model)}}
   Repeats []Block{{$model.Id}}Repeat
 {{end}}
 }
 
-func (self *Block{{.Model.Id}}) GetId() ModelId {
-  return {{.Model.Id}}
+func (self *Block{{$model.Id}}) GetId() ModelId {
+  return {{$model.Id}}
 }
 
-var model{{.Model.Id}} = smdx.ModelElement{
-	Id: {{.Model.Id}},
-	Name: "{{.Model.Name}}",
-	Length: {{.Model.Length}},
-	Blocks: []smdx.BlockElement{ {{range .Model.Blocks}}
+func init() {
+	smdx.RegisterModel(&smdx.ModelElement{
+	Id: {{$model.Id}},
+	Name: "{{$model.Name}}",
+	Length: {{$model.Length}},
+	Blocks: []smdx.BlockElement{ {{range $model.Blocks}}
 			smdx.BlockElement{ {{if gt (len .Name) 0 }}Name: "{{.Name}}",{{end}}
```

Comment:
Not that it's a huge deal, but the output with `BlockElement{Name: ""` is a bit odd